shell/platform/darwin/macos/framework/Source/FlutterVersions.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Cocoa/Cocoa.h>
+
+//-----------------------------------------------------------------------------
+/// @brief      Whether the operating system version is equal or higher than
+///             the given version.
+/// @param[in]  majorVersion            The major verison to compare. Must be
+///                                     a postive integer
+/// @param[in]  minorVersion            The minor verison to compare. Must be
+///                                     a postive integer if provided. If it is
+///                                     not provided, the function only compares
+///                                     up to major version.
+/// @param[in]  patchVersion            The patch verison to compare. Must be
+///                                     a postive integer if provided. If it is
+///                                     not provided, the function only compares
+///                                     up to minor version.
+bool isOperatingSystemAtLeastVersion(NSInteger majorVersion,
+                                     NSInteger minorVersion = -1,
+                                     NSInteger patchVersion = -1);
+
+//-----------------------------------------------------------------------------
+/// @brief      Whether the operating system version is equal or less than the
+///             given version.
+/// @param[in]  majorVersion            The major verison to compare. Must be
+///                                     a postive integer
+/// @param[in]  minorVersion            The minor verison to compare. Must be
+///                                     a postive integer if provided. If it is
+///                                     not provided, the function only compares
+///                                     up to major version.
+/// @param[in]  patchVersion            The patch verison to compare. Must be
+///                                     a postive integer if provided. If it is
+///                                     not provided, the function only compares
+///                                     up to minor version.
+bool isOperatingSystemAtMostVersion(NSInteger majorVersion,
```

Comment:
Why do we need this form? If we're putting an upper bound on a section of code, then we know the first OS version where that code is no longer needed. Why not just express everything that way and have a simpler API surface?

E.g., you have code that only runs for versions up through 10.13, which is equivalent to not running if it's 10.14+.